packages/fuels/src/cli.ts

```diff
@@ -1,27 +1,120 @@
+/* eslint-disable @typescript-eslint/triple-slash-reference */
+/// <reference path="./hbs.d.ts" />
+
 import { configureCliOptions as configureTypegenCliOptions } from '@fuel-ts/abi-typegen/cli';
 import { versions } from '@fuel-ts/versions';
 import { runVersions } from '@fuel-ts/versions/cli';
-import { Command } from 'commander';
+import { Command, Option } from 'commander';
+import { join } from 'path';
+
+import { build } from './cli/commands/build';
+import { deploy } from './cli/commands/deploy';
+import { dev } from './cli/commands/dev';
+import { init } from './cli/commands/init';
+import { withConfig } from './cli/commands/withConfig';
+import { withProgram } from './cli/commands/withProgram';
+import { Commands } from './cli/types';
+import { findPackageRoot } from './cli/utils/findPackageRoot';
+import { configureLogging } from './cli/utils/logger';
 
-export function run(argv: string[]) {
+export const onPreAction = (command: Command) => {
+  const opts = command.opts();
+  configureLogging({
+    isDebugEnabled: opts.debug,
+    isLoggingEnabled: !opts.silent,
+  });
+};
+
+export const configureCli = () => {
   const program = new Command();
 
   program.name('fuels');
-  program.version(versions.FUELS);
 
-  // routing `versions` sub-command
-  program
-    .command('versions')
-    .description('check for version incompatibilities')
-    .action(runVersions);
+  program.option('-D, --debug', 'Enables verbose logging', false);
+  program.option('-S, --silent', 'Omit output messages', false);
+
+  program.version(versions.FUELS, '-v, --version', 'Output the version number');
+  program.helpOption('-h, --help', 'Display help');
+  program.addHelpCommand('help [command]', 'Display help for command');
+
+  program.enablePositionalOptions(true);
+
+  program.hook('preAction', onPreAction);
+
+  /**
+   * Defining local commands
+   */
+
+  const pathOption = new Option('-p, --path <path>', 'Path to project root').default(process.cwd());
+
+  let command: Command;
+
+  const desc = `Relative path/globals to `;
+  const arg = `<path|global>`;
+
+  (command = program.command(Commands.init))
+    .description('Create a sample `fuel.config.ts` file')
+    .addOption(pathOption)
+    .option('-w, --workspace <path>', 'Relative dir path to Sway workspace')
```

Comment:
No, because you can use `workspace` or individual configs for `contracts`, `scripts`, and `predicates`.

But there's a problem with the validation. I'm looking into it.

```console
$ pnpm fuels init -o .                                                                                                                                 [12:35:28]
node:internal/streams/writable:287
  const state = stream._writableState;
                       ^

TypeError: Cannot read properties of undefined (reading '_writableState')
    at _write (node:internal/streams/writable:287:24)
    at Writable.write (node:internal/streams/writable:337:10)

Node.js v18.16.0
FAIL
```

Reply:
Fixed. https://github.com/FuelLabs/fuels-ts/pull/1291/commits/86b09cfc1ed5952ef196be688bf12ba22f25071e

```console
$ pnpm fuels init -o ./src/sway-programs-api
Error: Workspace not informed.
```